File: src/main/java/fr/ynerant/leveleditor/game/AutoTower.java
package fr.ynerant.leveleditor.game;

import java.util.Collection;

public class AutoTower extends Tower {
    public AutoTower(int x, int y) {
        super(x, y);
    }

    @Override
    public String getName() {
        return "autotower";
    }

    @Override
    public int getDamagePerShot() {
        return 20;
    }

    @Override
    public long getPeriod() {
        return 10;
    }

    @Override
    public int getPrice() {
        return 142;
    }

    @Override
    Collection<Mob> _filterDetectedMobs(Collection<Mob> mobs) {
        return mobs;
    }
}

File: src/main/scala/fr/ynerant/leveleditor/game/towers/AutoTower.scala
package fr.ynerant.leveleditor.game.towers

import fr.ynerant.leveleditor.game.mobs.Mob
import java.util


class AutoTower(override val x: Int, override val y: Int) extends Tower(x, y) {
	override def getName = "autotower"

	override def getDamagePerShot = 20

	override def getPeriod = 10

	override def getPrice = 142

	override private[towers] def _filterDetectedMobs(mobs: util.Collection[Mob]) = mobs
}
